### PR TITLE
fixes dotnet/msbuild#8958  TerminalLogger in .NET 8.0.100-preview.6 issues audible alerts on iTerm2

### PR DESCRIPTION
**Pull request by vlada-shubina**
fixes dotnet/msbuild#8958
### Context
TerminalLogger in .NET 8.0.100-preview.6 issues audible alerts on iTerm2
Iterm2 treats ;9 code differently from ConEmu: https://iterm2.com/documentation-escape-codes.html

### Changes Made
Disabled progress reporting on Mac

### Testing
Tested manually on Mac

### Notes

### CODE REVIEW

File: src/MSBuild/TerminalLogger/TerminalLogger.cs [hunk 1]
```diff
@@ -235,7 +236,12 @@ private void BuildStarted(object sender, BuildStartedEventArgs e)
 
         _buildStartTime = e.Timestamp;
 
-        Terminal.Write(AnsiCodes.SetProgressIndeterminate);
+        // dotnet/msbuild#8958: iTerm2 treats ;9 code to post a notification instead,
```
Comment (rainersigwald) [inline src/MSBuild/TerminalLogger/TerminalLogger.cs:239]: Nit: I prefer full URLs
```suggestion
        // https://github.com/dotnet/msbuild/issues/8958: iTerm2 treats ;9 code to post a notification instead,
```

File: src/MSBuild/TerminalLogger/AnsiCodes.cs [hunk 2]
```diff
@@ -95,13 +95,23 @@ internal static class AnsiCodes
     public const string ShowCursor = "\x1b[?25h";
 
     /// <summary>
-    /// Set progress state to a busy spinner.
+    /// Set progress state to a busy spinner. <br/>
+    /// Note: this code works only on ConEmu terminals, and conflicts with push a notification code on iTerm2.
```
Comment (rainersigwald) [inline src/MSBuild/TerminalLogger/AnsiCodes.cs:99]: This is over-broad; several terminals support it (including most importantly to our Windows folks, Windows Terminal). I might simplify to something like "should be wrapped with a support check".

File: src/MSBuild/TerminalLogger/TerminalLogger.cs [hunk 3]
```diff
@@ -235,7 +236,12 @@ private void BuildStarted(object sender, BuildStartedEventArgs e)
...
+        // so disable progress reporting on Mac.
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```
Comment (rainersigwald) [inline src/MSBuild/TerminalLogger/TerminalLogger.cs:241]: Maybe abstract this check to a method named something like `TerminalSupportsProgressReporting`, which this as the current implementation?

File: src/MSBuild/TerminalLogger/AnsiCodes.cs (hunk 2 above)
Comment (vlada-shubina) [reply src/MSBuild/TerminalLogger/AnsiCodes.cs:99]: is there any specification Windows Terminal adheres to?
I thought they adhere to ConEmu (at least to some extend).

File: .editorconfig [hunk 4]
```diff
@@ -15,6 +15,9 @@ trim_trailing_whitespace = true
 [project.json]
 indent_size = 2
 
+[*.txt]
+insert_final_newline = false
```
Comment (rainersigwald) [inline .editorconfig:19]: ❤️❤️❤️ Knew this setting had to be living _somewhere_.

**Review by rainersigwald — APPROVED** [review]
Having to split every validation test output in 3 is a bit unfortunate but this works for me.

. . . Hmm, actually, can we set the mock terminal in the tests to not emit this code consistently except for a specific test for this behavior? Up to you whether that seems worth it.

**Review by ladipro — APPROVED** [review]

**Comment by vlada-shubina** [comment]
> Having to split every validation test output in 3 is a bit unfortunate but this works for me.
> 
> . . . Hmm, actually, can we set the mock terminal in the tests to not emit this code consistently except for a specific test for this behavior? Up to you whether that seems worth it.

It depends on test intention actually. If the intention is to test real life e2e scenario, and Mac output is currently indeed different, it's better to keep them split as is. 

Unless we "unseal" `Terminal` implementation, and override only diff members, or make it configurable, the mock implementation can easily deviate and bug will be missed. Given that most of us is developing on Win, Mac and Linux issues can be unnoticed.

From my experience, it's worth to keep truly e2e scenarios unmocked. For specific tests, for example `DisplayNodesShowsCurrent`, `DisplayNodesOverwritesWithNewTargetFramework` it might be reasonable to use mock, however i don't have a full picture to make a decision for them.

File: src/MSBuild/TerminalLogger/Terminal.cs [hunk 5]
```diff
@@ -62,6 +63,12 @@ public int Width
         }
     }
 
+    /// <inheritdoc/>
+    /// <remarks>
+    /// https://github.com/dotnet/msbuild/issues/8958: iTerm2 treats ;9 code to post a notification instead, so disable progress reporting on Mac.
+    /// </remarks>
+    public bool SupportsProgressReporting { get; } = !RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
```
Comment (slang25) [inline src/MSBuild/TerminalLogger/Terminal.cs:70]:
```suggestion
    public bool SupportsProgressReporting { get; } = Environment.GetEnvironmentVariable("TERM_PROGRAM") != "iTerm.app";
```
If it's just iTerm2 perhaps this? (feels super hacky I know)

**Comment by vlada-shubina** [comment]
Unless there is more feedback on open items, this PR is ready to be merged.

File: src/MSBuild/TerminalLogger/Terminal.cs (hunk 5 above)
Comment (slang25) [reply src/MSBuild/TerminalLogger/Terminal.cs:70]: Just tried a few, and none of them appear to support it:
* Terminal (built in)
* iTerm2
* Hyper
* Alacritty

**Pull request merged**